runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -539,6 +577,17 @@ impl<T: Trait> Module<T> {
             return Err(Error::<T>::StakeShouldBeEmpty.into());
         }
 
+        if let Some(execution_block) = exact_execution_block {
+            if execution_block == Zero::zero() {
+                return Err(Error::<T>::ZeroExactExecutionBlock.into());
+            }
+
+            let now = Self::current_block();
+            if execution_block <= now + parameters.grace_period {
```

Comment:
This does not match the handbook:

https://joystream.gitbook.io/joystream-handbook/governance/proposals#submit-proposal

Reply:
Fixed - changed the check and the algorithm.